builtin/logical/aws/path_static_creds.go

```diff
@@ -0,0 +1,99 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	pathStaticCreds = "static-creds"
+
+	paramAccessKeyID      = "access_key_id"
+	paramSecretsAccessKey = "secret_access_key"
+)
+
+type awsCredentials struct {
+	AccessKeyID     string `json:"access_key_id" structs:"access_key_id" mapstructure:"access_key_id"`
+	SecretAccessKey string `json:"secret_access_key" structs:"secret_access_key" mapstructure:"secret_access_key"`
```

Comment:
Wanted to note that other Vault static role rotation features will usually include some additional information when reading credentials. For example, the TTL until next rotation and last time it was rotated (example [db static roles](https://developer.hashicorp.com/vault/api-docs/secret/databases#get-static-credentials), [ldap static roles](https://developer.hashicorp.com/vault/api-docs/secret/ldap#static-role-passwords)). Might be a nice addition if it's not too much of a lift.

Not blocking and we can let this be driven by feedback 🙂

Reply:
I'll add it to the improvements list!